### PR TITLE
Moved protos to dedicated directory and updated the proto.

### PR DESCRIPTION
**Pull request by gavinlinasd**
Moved protos to dedicated directory so it can be used by client.

- Added proto directory and moved protocached.proto there
- Added BUILD file for proto dir
- Added more layers of proto message for the API
- Ran the formatter.

### CODE REVIEW

File: src/proto/BUILD [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        "protocached.proto",
+    ],
+)
+
+cc_proto_library(
+    name = "protocached_cc_proto",
+    deps = [
+        ":protocached_proto"
+    ],
+    visibility = ["//visibility:public"],
```
Comment (joejoevictor) [inline src/proto/BUILD]: Should we just make the whole package public instead of just this rule?

File: src/proto/BUILD [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+
```
Comment (joejoevictor) [inline src/proto/BUILD]: Nit, 2 empty lines? Maybe remove 1?

File: src/proto/protocached.proto [hunk 3]
```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package protocached;
+
+// Result for rpc response
+message ResponseResult {
+  enum ResultType {
```
Comment (joejoevictor) [inline src/proto/protocached.proto]: Maybe call this ResultStatus instead?

File: src/proto/protocached.proto [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    SUCCESS = 0;
+    ERROR = 1;
+  }
+
+  // specify the response result
+  ResultType result = 1;
```
Comment (joejoevictor) [inline src/proto/protocached.proto]: Maybe change the name of the field to result_type or result_status  if you decide to change the message type name to ResultStatus?

File: src/proto/protocached.proto [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  // optional error message if the result is ERROR
+  string errorMsg = 2;
```
Comment (joejoevictor) [inline src/proto/protocached.proto]: I thought the style guide standard is to use snake case for proto field name?

Also, shouldn't this be optional? If the result status is successful, there wouldn't be any error message, right?

Comment (gavinlinasd) [reply src/proto/protocached.proto]: Yes, this is (and should be) optional. AFAIK, i think proto3 syntax default everything to be optional if you don't specify (at least within Google it's true), I will double check that.

Comment (gavinlinasd) [reply src/proto/protocached.proto]: Yeah, in proto3 syntax, everything is optional (https://developers.google.com/protocol-buffers/docs/proto3#specifying-field-rules). They get rid of required/optional field. You can either be optional or repeated.

Also see: http://www.crankuptheamps.com/blog/posts/2017/10/12/protobuf-battle-of-the-syntaxes/

File: src/client/BUILD [hunk 6]
```diff
@@ -1,3 +1,5 @@
+# BUILD file for ProtoCached client
+package(default_visibility = ["//visibility:public"])
```
Comment (joejoevictor) [inline src/client/BUILD]: Nit again...Empty line after this line? Maybe we should just enable some sort of formatter, I feel this happens quite a lot.

File: src/proto/BUILD [hunk 7]
```diff
@@ -1,4 +1,5 @@
 # BUILD file for protos
+package(default_visibility = ["//visibility:public"])
```
Comment (joejoevictor) [inline src/proto/BUILD]: Ditto

**Comment by joejoevictor** [comment]
Gonna merge this in.

**Pull request merged**